### PR TITLE
MON-11757 fix stickyness for ack

### PR DESCRIPTION
**Pull request by tanguyvda**
## Description

resource status page uses the api/beta/monitoring/resources/acknowledge  endpoint
this endpoint set the sticky value to 1 or 0 instead of 2 or 0 according to this https://assets.nagios.com/downloads/nagioscore/docs/externalcmds/cmdinfo.php?command_id=40

Meaning that if you set the stickyness in your default configuration, it will not work at all and set the ack like it wasn't sticky

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 20.04.x
- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

without this PR,
- go in administration -> parameters -> monitoring menu and tick the sticky checkbox  (meaning ack are going to be sticky)
-  go in the monitoring -> resources status page and put an ack on a service or a host, put the following comment "default sticky value: 1)
- wait for your ack to be displayed in the monitoring (yellow background on the service or host row)
- when ack is displayed, in your central database, run the following query : 
`select sticky,comment_data from centreon_storage.acknowledgements;` 
- you should have the following result: `|      0 | default sticky 1`

with this PR,
- repeat the above steps
- you should have the following result: `|      1 | default sticky 1`

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Domain/Engine/EngineService.php [hunk 1]
```diff
@@ -98,10 +98,15 @@ public function addHostAcknowledgement(Acknowledgement $acknowledgement, Host $h
             throw new EngineException('Host name can not be empty');
         }
 
+        $sticky = 0;
+        if ((int) $acknowledgement->isSticky() === 1) {
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/Engine/EngineService.php:102]:
```suggestion
        if ($acknowledgement->isSticky() === true) {
```

File: src/Centreon/Domain/Engine/EngineService.php [hunk 2]
```diff
@@ -124,11 +129,16 @@ public function addServiceAcknowledgement(Acknowledgement $acknowledgement, Serv
             throw new EngineException('The host of service is not defined');
         }
 
+        $sticky = 0;
+        if ((int) $acknowledgement->isSticky() === 1) {
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/Engine/EngineService.php:133]:
```suggestion
        if ($acknowledgement->isSticky() === true) {
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by sc979 — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

File: src/Centreon/Domain/Engine/EngineService.php [hunk 3]
```diff
@@ -98,10 +98,15 @@ public function addHostAcknowledgement(Acknowledgement $acknowledgement, Host $h
             throw new EngineException('Host name can not be empty');
         }
 
+        $sticky = 0;
+        if ($acknowledgement->isSticky() === true) {
+            $sticky = 2;
+        }
+
```
Comment (callapa) [inline src/Centreon/Domain/Engine/EngineService.php:105]:
```suggestion
```

File: src/Centreon/Domain/Engine/EngineService.php [hunk 4]
```diff
@@ -98,10 +98,15 @@ public function addHostAcknowledgement(Acknowledgement $acknowledgement, Host $h
...
         $preCommand = sprintf(
             'ACKNOWLEDGE_HOST_PROBLEM;%s;%d;%d;%d;%s;%s',
             $host->getName(),
-            (int) $acknowledgement->isSticky(),
+            $sticky,
```
Comment (callapa) [inline src/Centreon/Domain/Engine/EngineService.php:109]:
```suggestion
            $acknowledgement->isSticky() ? 2 : 0,
```

File: src/Centreon/Domain/Engine/EngineService.php [hunk 5]
```diff
@@ -124,11 +129,16 @@ public function addServiceAcknowledgement(Acknowledgement $acknowledgement, Serv
...
+        if ($acknowledgement->isSticky() === true) {
+            $sticky = 2;
+        }
+
         $preCommand = sprintf(
             'ACKNOWLEDGE_SVC_PROBLEM;%s;%s;%d;%d;%d;%s;%s',
             $service->getHost()->getName(),
             $service->getDescription(),
-            (int) $acknowledgement->isSticky(),
+            $sticky,
```
Comment (callapa) [inline src/Centreon/Domain/Engine/EngineService.php:141]:
```suggestion
            $acknowledgement->isSticky() ? 2 : 0,
```

File: src/Centreon/Domain/Engine/EngineService.php [hunk 6]
```diff
@@ -124,11 +129,16 @@ public function addServiceAcknowledgement(Acknowledgement $acknowledgement, Serv
             throw new EngineException('The host of service is not defined');
         }
 
+        $sticky = 0;
+        if ($acknowledgement->isSticky() === true) {
+            $sticky = 2;
+        }
+
```
Comment (callapa) [inline src/Centreon/Domain/Engine/EngineService.php:136]:
```suggestion
```

**Comment by omercier** [comment]
Internal reference: MON-11757

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by callapa — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Comment by callapa** [comment]
Already fixed in the develop branch. Must be cherry-picked in the others.

**Comment by callapa** [comment]
https://github.com/centreon/centreon/pull/10432

**Pull request closed without merging**